test/zdtm/static/memfd04.c

```diff
@@ -73,20 +74,27 @@ static const size_t script_len = sizeof(script) - 1;
 
 int main(int argc, char *argv[])
 {
+	char *addr;
 	int fd;
 
 	test_init(argc, argv);
 
 	fd = _memfd_create("somename", 0);
 	if (fd < 0) {
-		fail("memfd_create()");
+		pr_perror("memfd_create()");
+		return 1;
+	}
+	if (ftruncate(fd, script_len) == -1) {
+		pr_perror("ftruncate");
 		return 1;
 	}
 
-	if (write(fd, script, script_len) != script_len) {
-		fail("write(memfd)");
+	addr = mmap(NULL, PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_FILE | MAP_SHARED, fd, 0);
+	if (addr == MAP_FAILED) {
+		pr_perror("mmap");
 		return 1;
 	}
+	memcpy(addr, script, script_len);
 
 	if (!test_exec_fd(fd))
```

Comment:
I'm surprised this actually works to execute a fd that is RW-mmapped in another process. Can we make this a separate test in case this changes?

Reply:
Yes, we can. I am not sure that it is really worth the separate test...

Reply:
It is testing a different conditions, so I'd prefer to have a separate test for better bug signal. We could extract some common code if that's a concern.

Reply:
Strictly speaking, it tests the same major condition is that an origin memfd is preserved when it is possible.

Reply:
I have added memfd05...